testutils/src/main/java/com/microsoft/identity/internal/testutils/MockHttpResponse.java

```diff
@@ -22,21 +22,73 @@
 // THE SOFTWARE.
 package com.microsoft.identity.internal.testutils;
 
+import android.util.Pair;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.microsoft.identity.common.internal.net.HttpClient.HttpMethod;
 import com.microsoft.identity.common.internal.net.HttpResponse;
 
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
 /**
- * Class to set a mock http response at runtime.
+ * Class to  set a mock request interceptor at runtime.
  */
 public class MockHttpResponse {
 
-    private static HttpResponse sHttpResponse;
+    private static final HashMap<Pair<HttpMethod, String>, MockHttpRequestInterceptor> interceptors = new HashMap<>();
+
+    public static void setInterceptor(@NonNull final MockHttpRequestInterceptor interceptor) {
+        MockHttpResponse.setInterceptor(interceptor, null, ".*");
+    }
+
+    public static void setInterceptor(@NonNull final MockHttpRequestInterceptor interceptor,
+                                      @NonNull String urlRegex) {
+        MockHttpResponse.setInterceptor(interceptor, null, urlRegex);
+    }
+
+    public static void setInterceptor(@NonNull final MockHttpRequestInterceptor interceptor,
+                                      @NonNull HttpMethod method) {
+        MockHttpResponse.setInterceptor(interceptor, method, ".*");
+    }
 
 
-    public static HttpResponse getHttpResponse() {
-        return sHttpResponse;
+    public static void setInterceptor(@NonNull final MockHttpRequestInterceptor interceptor,
+                                      @Nullable final HttpMethod method,
+                                      @NonNull final String urlRegex) {
+        interceptors.put(new Pair<>(method, urlRegex), interceptor);
+    }
+
+    public static MockHttpRequestInterceptor intercept(@NonNull final HttpMethod method, @NonNull final URL url) {
+        for (Pair<HttpMethod, String> pair : interceptors.keySet()) {
+            if ((pair.first == null || pair.first.compareTo(method) == 0) && url.toString().matches(pair.second)) {
+                return interceptors.get(pair);
+            }
+        }
+        return null;
```

Comment:
Can you add some comments describing what is happening here

Reply:
Yes, sure.